test/features/transaction/transaction_details_page_test.dart

```diff
@@ -0,0 +1,131 @@
+
+void main() {
+  group('TransactionDetailsPage', () {
+    final sendTransaction = TestData.getTransaction();
+    final depositTransaction =
+        TestData.getTransaction(type: TransactionType.deposit);
+    final withdrawTransaction =
+        TestData.getTransaction(type: TransactionType.withdraw);
+
+    const mockTransactionNotifierWithSendTransaction =
+        MockTransactionNotifierWithData(
+      transactionType: TransactionType.send,
+    );
+    const mockTransactionNotifierWithError = MockTransactionNotifierWithError();
+
+    late MockTransactionNotifier mockSendTransactionNotifier;
+    late MockTransactionNotifier mockDepositTransactionNotifier;
+    late MockTransactionNotifier mockWithdrawTransactionNotifier;
+    late MockTransactionNotifier nullMockTransactionNotifier;
+    late MockTransactionNotifier erroringMockTransactionNotifier;
+
+    setUp(() {
+      mockSendTransactionNotifier =
+          MockTransactionNotifier(() => sendTransaction);
+      mockDepositTransactionNotifier =
+          MockTransactionNotifier(() => depositTransaction);
+      mockWithdrawTransactionNotifier =
+          MockTransactionNotifier(() => withdrawTransaction);
+      nullMockTransactionNotifier = MockTransactionNotifier();
+      erroringMockTransactionNotifier = MockTransactionNotifier(
+        () => throw StateError('Error loading transaction'),
+      );
+    });
+
+    Widget transactionDetailsTestWidget({
+      required MockTransactionNotifierType mockTransactionNotifierType,
+    }) =>
+        WidgetHelpers.testableWidget(
+          child: TransactionDetailsPage(
+            pfi: TestData.getPfi('did:dht:pfiDid'),
+            exchangeId: 'rfq_01ha835rhefwmagsknrrhvaa0k',
+          ),
+          overrides: [
+            transactionProvider.overrideWith(
+              () => switch (mockTransactionNotifierType) {
+                MockTransactionNotifierWithData() => switch (
+                      mockTransactionNotifierType.transactionType) {
+                    TransactionType.send => mockSendTransactionNotifier,
+                    TransactionType.deposit => mockDepositTransactionNotifier,
+                    TransactionType.withdraw => mockWithdrawTransactionNotifier,
+                  },
+                MockTransactionNotifierWithNullData() =>
+                  nullMockTransactionNotifier,
+                MockTransactionNotifierWithError() =>
+                  erroringMockTransactionNotifier,
+              },
+            ),
+          ],
+        );
+
+    testWidgets('should show correct payout and payin amounts', (tester) async {
+      await tester.pumpWidget(
+        transactionDetailsTestWidget(
+          mockTransactionNotifierType:
+              mockTransactionNotifierWithSendTransaction,
+        ),
+      );
+      await tester.pumpAndSettle();
+
+      expect(
+        find.text('100.01'),
+        findsOneWidget,
+      ); // Adjust to your currency setup.
```

Comment:
[comments in test file not required](https://github.com/TBD54566975/didpay/pull/312#discussion_r1797253745)